src/test/java/org/dswarm/graph/xml/test/XMLResourceTest.java

```diff
@@ -0,0 +1,140 @@
+import com.sun.jersey.multipart.MultiPart;
+
+/**
+ * @author tgaengler
+ */
+public abstract class XMLResourceTest extends BasicResourceTest {
+
+	private static final Logger	LOG	= LoggerFactory.getLogger(XMLResourceTest.class);
+
+	// we need PNX gson
+	private static final String DEFAULT_GDM_FILE_NAME = "test-pnx.gson";
+
+	public XMLResourceTest(final Neo4jDBWrapper neo4jDBWrapper, final String dbTypeArg) {
+
+		super(neo4jDBWrapper, "/xml", dbTypeArg);
+	}
+
+	@Test
+	public void readPNXXMLFromDB() throws IOException {
+
+		LOG.debug("start read test for GDM resource at " + dbType + " DB");
+
+		final String dataModelURI = "http://data.slub-dresden.de/resources/1";
+		final String recordClassURI = "http://www.exlibrisgroup.com/xsd/primo/primo_nm_bib#recordType";
+		final String recordTag = "http://www.exlibrisgroup.com/xsd/primo/primo_nm_bib#record";
+
+		writeGDMToDBInternal(dataModelURI, DEFAULT_GDM_FILE_NAME);
+
+		readXMLFromDB(recordClassURI, dataModelURI, Optional.<String>absent(), Optional.of(recordTag), Optional.<Integer>absent(), "test-pnx.xml");
+
+		LOG.debug("finished read test for GDM resource at " + dbType + " DB");
+	}
+
+	private void readXMLFromDB(final String recordClassURI, final String dataModelURI, final Optional<String> optionalRootAttributePath, final Optional<String> optionalRecordTag, final Optional<Integer> optionalVersion, final String expectedFileName) throws IOException {
+
+		final ObjectMapper objectMapper = Util.getJSONObjectMapper();
+		objectMapper.enable(SerializationFeature.INDENT_OUTPUT);
+		final ObjectNode requestJson = objectMapper.createObjectNode();
+
+		requestJson.put(DMPStatics.RECORD_CLASS_URI_IDENTIFIER, recordClassURI);
+		requestJson.put(DMPStatics.DATA_MODEL_URI_IDENTIFIER, dataModelURI);
+
+		if(optionalRootAttributePath.isPresent()) {
+
+			requestJson.put(DMPStatics.ROOT_ATTRIBUTE_PATH_IDENTIFIER, optionalRootAttributePath.get());
+		}
+
+		if(optionalRecordTag.isPresent()) {
+
+			requestJson.put(DMPStatics.RECORD_TAG_IDENTIFIER, optionalRecordTag.get());
+		}
+
+		if(optionalVersion.isPresent()) {
+
+			requestJson.put(DMPStatics.VERSION_IDENTIFIER, optionalVersion.get());
+		}
+
+		final String requestJsonString = objectMapper.writeValueAsString(requestJson);
+
+		// POST the request
+		final ClientResponse response = target().path("/get").type(MediaType.APPLICATION_JSON_TYPE).accept(MediaType.APPLICATION_XML_TYPE)
+				.post(ClientResponse.class, requestJsonString);
+
+		Assert.assertEquals("expected 200", 200, response.getStatus());
+		Assert.assertEquals(MediaType.APPLICATION_XML_TYPE, response.getType());
+
+		final String actualXML = response.getEntity(String.class);
+
+		Assert.assertNotNull(actualXML);
+
+		// compare result with expected result
+		final URL expectedFileURL = Resources.getResource(expectedFileName);
+		final String expectedXML = Resources.toString(expectedFileURL, Charsets.UTF_8);
+
+		Assert.assertEquals(expectedXML, actualXML);
```

Comment:
ok, should be fixed right now by utilising XMLUnit (instead of a simple string comparison).